test/model/src/driver/kv_workload.cpp

```diff
@@ -202,6 +202,46 @@ parse(const char *str)
       "Cannot parse operation: Unknown operation \"" + std::string(name) + "\"");
 }
 
+/*
+ * transactional --
+ *     Check if the workload operation is a transactional operation, including begin and commit.
+ */
+bool
+transactional(const any &op) noexcept
+{
+    return std::holds_alternative<begin_transaction>(op) ||
+      std::holds_alternative<commit_transaction>(op) || std::holds_alternative<insert>(op) ||
+      std::holds_alternative<prepare_transaction>(op) || std::holds_alternative<remove>(op) ||
+      std::holds_alternative<rollback_transaction>(op) ||
+      std::holds_alternative<set_commit_timestamp>(op) || std::holds_alternative<truncate>(op);
```

Comment:
Sorry to bring this up again. But this still makes me a little nervous - it'd be very easy to add a new type and forget to update it here (we don't get help from the compiler like a `switch` would).

Even though it's a little clunkier, perhaps we could do something like:
```
if (std::holds_alternative<begin_transaction>(op) || ...)
    return true;
else if (std::holds_alternative<create_table> || ...)
    // non-txn stuff
    return false;
throw model_exception("...");
```

So if someone forgets to add whether or not a new type is transactional, it'll be a nice loud failure.

Reply:
Yes, explicit checks here make sense. I tried a different approach using inheritance in [453d758](https://github.com/wiredtiger/wiredtiger/pull/10318/commits/453d7582de89866839710278d41de3f0bd08423f), where the compiler would catch this.